ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d.cpp

```diff
@@ -35,8 +35,10 @@ Tensor MaxPoolNewOp::operator()(uint8_t queue_id, const Tensor& input_tensor, ui
 
     if (!memory_config.shard_spec.has_value()) {
         // Input is not sharded. Perform sharding.
+        conv2d::Conv2dConfig config;
+        config.shard_layout = TensorMemoryLayout::HEIGHT_SHARDED;
         parallel_config = conv2d::determine_parallel_config(
-                                            true,
+                                            config,
```

Comment:
I would still prefer just passing in the shard scheme directly instead of the whole config. But up to you :)